### ISSUE TITLE
StaxRip does not display file names in Source or Target dialog, "Failed to load source" when clicking Crop or Preview

### ISSUE THREAD

**Issue by burnchar** [staxrip/staxrip#238]
**Describe the bug**
When I open a file -- any video file with any format or container -- Staxrip shows no file name and no error. If I click the "Crop" or "Preview" menu items, I get, "Failed to load source" modal dialog box.
If I click, "Next", the video **does** indeed show up in the jobs list and does create a log file.
I can click "Start" and the video does encode. However, I cannot reasonably set many options such as resolution because Staxrip does not know what the source video resolution is. This makes Staxrip nearly unusable.

This has been true for the last three releases. It might be part of an included utility application, but I have reinstalled Staxrip and AVISynth etc. many times.

**Expected behavior**
When I open a video, the file name should appear in the "Source" text field. I should be able to change and save encoding options. I should see the file name, e.g. c:\foo\foo_new.mkv" in the "Target" text field.

**How to reproduce the issue**

This happens any time I encode any video of any kind with Staxrip.

[metadata.zip](https://github.com/staxrip/staxrip/files/4801215/metadata.zip)

**Comment by stax76** [comment]
It's messy due to avisynth having introduced many compatibility issues lately.

You need avisynth 361test8, copy the included one to system32.

**Comment by 44vince44** [comment]
> 
> 
> It's messy due to avisynth having introduced many compatibility issues lately.
> 
> You need avisynth 361test8, copy the included one to system32.

@burnchar the version you need to use is from latest beta of staxrip, get it [here](https://www.dropbox.com/sh/4ctl2y928xkak4f/AAADEZj_hFpGQaNOdd3yqcAHa?dl=0)
Then open the folder (in staxrip tree) Apps\FrameServer\AviSynth\
There you find two dlls, copy them to Windows\System32 if you g=have Avisynth installed on your Windows system. If not, just use the beta directly.
This is the version of Avisynth that fixes many bugs. Please try and tell us if everything works

**Comment by burnchar** [comment]
tl;dr: Yes, it works.

Detail:
- AviSynth 3.6.1 is now out. It does not work, even though the DLL included in the beta linked above is pretty much a pre-release 3.6.1. Stated differently, the beta DLL in \system32 works following 44vince44's instructions, but when I install AviSynth+ 3.6.1 released June 20 2020, and StaxRip uses its DLLs, the bug returns.

- I tried the beta. It has the problem reported above, even when I verified that it and the current release version are tied to the same avisynth.dll in \system32. This might not be the beta, but might be related to the directory name weirdness described below.

- StaxRip 2.1.3.0 with the beta DLLs works, as long as I do not change its directory (see below).

- For some reason, if I rename the directory of StaxRip 2.1.3.0 (current release) or copy it to a new directory from its current "C:\StaxRip-lessold", e.g. to "C:\StaxRip", it exhibits the problem in my bug report. I have verified both copies point to the exact same AviSynth version in \system32 (as they should, being literal copies). I did this several times. I don't know how this could possibly be true.

Thank you for the help. Let me know if I can help in any way.

**Comment by 44vince44** [comment]
Can you tell me what the metadata.zip shows? I opened the log and it seemed a successful operation

**Comment by 44vince44** [comment]
I'm not sure anymore it's related to Avisynth. Maybe that was a quick assumption.
I am trying to reproduce the issue, and I can't reproduce it. You say the problem happens with any video?

**Comment by 44vince44** [comment]
Here is a test you could do:

Can you try the following
1. check in the Apps > Manage menu > Tools > Check status of all required tools. 
   If it doesn't say OK, don't continue and report please
  
2.Copy and Rename Warframe Railjack 2020.06.18.mp4's copy to testffvid.mp4
  (so we're sure you can't have a temp folder with this name)
3. before drag and drop, in the filters section, first filter "Source", right click and select Source > FFVideoSource
4. now drag and drop and see what happens

**Comment by stax76** [comment]
You can try clear custom paths in the apps dialog and restart staxrip.

**Comment by burnchar** [comment]
I found the cause!

But first:

- Yes, I always deleted the _temp directory prior to trying an encode

- "Apps > Manage menu > Tools > Check status of all required tools" always says "OK!", except for -stable I have to configure the AVISynth version number to proceed (with the system32 beta avisynth files).

- Clear Custom Paths never did anything because I had no custom paths defined in any version.


Moving on:

1. I tried many permutations of StaxRip versions, AVISynth changes, etc. and found that the one common factor no matter the versions is the directory "C:\StaxRip"; use of any _other_ directory succeeds.

2. The only thing in common had to be the registry, so I searched it for instances of "c:\staxrip\".

3. I found many references, but bottom line is that the folder is configured to use a different configuration files directory than the others (see image).

4. **When I rename that directory, StaxRip in c:\staxrip asks for a new config folder and works without issue**. Therefore, something in the configuration file directory inherited from previous versions is causing the problem.

StaxRip seems unable to use the old config directory, or perhaps it is somehow corrupt, but I have attached that directory's entire contents to help reproduce the issue.

[users-charles-appdata-local-staxripx64.zip](https://github.com/staxrip/staxrip/files/4809880/users-charles-appdata-local-staxripx64.zip)




![image](https://user-images.githubusercontent.com/681802/85230537-22ef2200-b3ae-11ea-9bcd-c4a92ce81a22.png)

**Comment by stax76** [comment]
I can reproduce it but so far no luck finding the cause.

**Comment by Dendraspis** [comment]
I would bet StaxRip was executed without neccessary permissions on C:\StaxRip - for example run as "User" where you have restricted write permissions on C:\. Then the program folder was selected for the settings and because StaxRip had no permission to write in C:\StaxRip, Windows chose the local appdata folder instead to let it run.

**Comment by stax76** [comment]
All the templates got corrupted, they all have a boolean field enabled that should only get enabled when projects are added to the job list for batch processing. I have absolutely no idea how this field could end up enabled in a template.

It was however easy to reset this field after template loading so it should work in the next build.

But since I don't understand how it happened, I can only hope that it's completely solved. I'll upload a new build soon.

**Comment by Dendraspis** [comment]
@burnchar 
Did you use Batch File import recently before this problem occurs?

**Comment by 44vince44** [comment]
@stax76 when you see all the backups of the settings, this folder has been the same since v1.6.0.0.
It is probable that templates had been saved once with a buggy version of Staxrip, and that boolean field got enabled and remained so afterwards.

**Comment by stax76** [comment]
New beta is online.

**Comment by burnchar** [comment]
> I have absolutely no idea how this field could end up enabled in a template.

This would have happened in release 2.0.8.0. I use only 2 or 3 templates, so did not even open all of them. I am not sure how this might have happened.


> Did you use Batch File import recently before this problem occurs?
I am not sure exactly what "batch file import" references. I regularly batch encodes (one output per file) and regularly have multiple instances work on the batch. 

> I would bet StaxRip was executed without neccessary permissions on C:\StaxRip
Nope, I chose c:\Staxrip because, across certain versions, Staxrip did not like being in the Program FIles directory. I deleted "C:\Staxrip" entirely before each "reinstall", and have no special permission inheritance set up.

> It is probable that templates had been saved once with a buggy version of Staxrip
This tracks, though I cannot imagine a reason that any version of StaxRip would re-save each and every template except perhaps a file format change.

@stax76 @44vince44 , @Dendraspis: Thank you for the new release. I appreciate your efforts. I agree that this is now closed.
